app/src/main/kotlin/com/wire/android/migration/feature/MigrateClientsDataUseCase.kt

```diff
@@ -74,6 +78,29 @@ class MigrateClientsDataUseCase @Inject constructor(
             }
         }
 
+    private fun fixSessionFileNames(proteusDir: File, domain: String) {
+        val sessionsDir = File(proteusDir, "sessions")
+        if (sessionsDir.exists() && sessionsDir.isDirectory) {
+            sessionsDir.listFiles { file -> !file.isDirectory }?.forEach { session ->
+                val fixedSessionFileName = fixSessionFileName(session.name, domain)
```

Comment:
this will still not work in a federated environment, for that one we need to get the domain of each user to fix the name

Reply:
maybe a check if federation is enabled for the user first
if no then using the user self domain is 100% safe
otherwise we need to quarry the domain of each user (the user id will be the file name) form the scala app DB

Reply:
ok I updated it, for federated environment now it gathers ids of all users related to session files and fetches data from the scala db to get a domain of every user and rename the file